weave/config.py

```diff
@@ -128,3 +131,37 @@ def get_mongo_db():
             "mongodb", username="root", password="example"
         )
     return client
+
+class MongoNames(Enum):
```

Comment:
MongoDB will be used or created in different places within weave. Therefore, global variables can be referred throughout weave. Global variables should be constant, immutable, and a singleton. 

Constant: the reference cannot be re-assigned or dereferenced.
Immutable: the object cannot be modified.
Singleton: there is only one object created of the class.

At runtime, we are guaranteed that the reference will not be modified.  

ex.
from enum import Enum 

class HTTPCode(Enum):
    BAD_REQUEST = 400
    NOT_FOUND = 404
    BAD_GATEWAY = 502

print(HTTPCode.BAD_REQUEST.name) <---BAD_REQUEST
print(HTTPCode.BAD_REQUEST.value) <---400
HTTPCode.BAD_REQUEST.value = 100 <---AttributeError: <enum 'Enum'> cannot set attribute 'value'

example without using Enum:

def evaluate_response(response: HTTPResponse) -> str:
    if response.code() == 404:
        return "Not Found"
    elif response.code() == 502:
        return "???"
    elif response.code() == 400:
        return "???"
    else:
        return "Unknown Status Code"

using enums:
def evaluate_response(response: HTTPResponse) -> str:
    if response.code() == HTTPCode.NOT_FOUND:
        return "Not Found"
    elif response.code() == HTTPCode.BAD_GATEWAY:
        return "???"
    elif response.code() == HTTPCode.BAD_REQUEST:
        return "???"
    else:
        return "Unknown Status Code"


"This PEP proposes adding an enumeration type to the Python standard library.
An enumeration is a set of symbolic names bound to unique, constant values. Within an enumeration, the values can be compared by identity, and the enumeration itself can be iterated over." (https://peps.python.org/pep-0435/)

"An enumeration: is a set of symbolic names (members) bound to unique values
can be iterated over to return its canonical (i.e. non-alias) members in definition order
uses call syntax to return members by value
uses index syntax to return members by name" (https://docs.python.org/3/library/enum.html)

Reply:
I think this enum is being used exclusively by the mongo class though, right? If so, then it would be less complicated to just stick them in as constants inside that class IMO